src/oracle/connection/stmt.rs

```diff
@@ -770,6 +745,8 @@ pub unsafe extern "C" fn cbf_get_data(octxp: *mut c_void,
         }
     }
 
+    // TODO: somehow find a solution to allocate the right buffer size here
+    // maybe just matching on the requested output type?
     // Provide the address of the storage where the data is to be returned
     ctx.store.resize(NUM_ELEMENTS, 0);
```

Comment:
I think we should change the buffer size here corresponding to the bind type. The only solution I'm aware of is just matching on the bind type and then set a buffer size.
A probably better solution would be to get these things from oci. Maybe it is possible to do something similar to `get_attr_type_and_size`?

Reply:
This needs to be the size of the type registered with `OCIBindByPos` above.
Couldn't we allocate the buffer in the `for tpe in metadata {` loop, since there we know the type? Otherwise we'd need to store the type info and/or buffer size in the `BindContext`.

Reply:
@weiznich `get_attr_type_and_size` does nothing magically. Actually we derive from the type the size. The only part where this would matter are again strings, but this can be "easily" solved, since the callback is as long there is data. So we'd set `value_sz` to e.g. 2GiB, but allocate just, e.g. 2MiB, for our buffer. If the callback is executed a second time, we increase the size of the vector.

Reply:
@weiznich `get_attr_type_and_size` derives internally from a type the size. So it's hardcoded anyway. However we could put this in single method and therefore use it here again.

(I made a similar comment already, but can't see it)